Habitica/src/main/java/com/habitrpg/android/habitica/data/implementation/TaskRepositoryImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.habitrpg.android.habitica.data.implementation;
+
+import com.habitrpg.android.habitica.APIHelper;
+import com.habitrpg.android.habitica.R;
+import com.habitrpg.android.habitica.data.TaskRepository;
+import com.habitrpg.android.habitica.data.local.TaskLocalRepository;
+import com.habitrpg.android.habitica.ui.helpers.UiUtils;
+import com.magicmicky.habitrpgwrapper.lib.api.ApiService;
+import com.magicmicky.habitrpgwrapper.lib.models.TaskDirection;
+import com.magicmicky.habitrpgwrapper.lib.models.TaskDirectionData;
+import com.magicmicky.habitrpgwrapper.lib.models.responses.HabitResponse;
+import com.magicmicky.habitrpgwrapper.lib.models.tasks.Task;
+import com.magicmicky.habitrpgwrapper.lib.models.tasks.TasksOrder;
+
+import java.util.ArrayList;
+
+import rx.Observable;
+
+
+public class TaskRepositoryImpl extends BaseRepositoryImpl<TaskLocalRepository> implements TaskRepository {
+
+
+    private APIHelper apiHelper;
```

Comment:
yeah I guess it would be better to use only the `apiclient` , but I'd like to implement the same interface of `ApiService` which would result that `configureApiCallObserver`  has to return `HabitResponse<T>` again

or we'd have two interfaces one with `HabitResponse<T>` and one without, what do you think? @vIiRuS